Test.cpp

```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+#include <stdexcept>
+
+class FactorialCalculator {
+private:
+    
+public:
+    // Method to calculate the factorial of a number
+    static long long calculate(long long number) {
+        if (number < 0) {
+            throw std::invalid_argument("Number must be non-negative.");
+        }
+
+        return number == 0 ? 1 : number * calculate(number - 1);
+    }
```

Comment:
Use a more descriptive name for this method, such as `calculateFactorial`, to enhance readability.